### PR TITLE
fix(dashboard-filters): Allow saving with All Projects

### PR DESCRIPTION
**Pull request by narsaynorath**
Dashboards weren't handling the -1, All Projects, constant properly when
saving. When the request is sent with projects=[-1], we'll set a boolean
in the filters property called `all_projects`. When updating the
dashboard after setting this, we need to check if it should be
overridden

How to reproduce the bug:
1. Set a dashboard's project filter to "All Projects"
2. Hit "Save"
3. There should be an error toast saying "A dashboard with this title already exists"

### CODE REVIEW

File: src/sentry/api/serializers/rest_framework/dashboard.py [hunk 1]
```diff
@@ -324,10 +325,16 @@ def update_dashboard_filters(self, instance, validated_data):
...
 
+        filters = {}
+
         if "projects" in validated_data:
-            instance.projects.set(validated_data["projects"])
+            if validated_data["projects"] == ALL_ACCESS_PROJECTS:
+                filters["all_projects"] = True
+            else:
+                if instance.filters and instance.filters.get("all_projects"):
+                    instance.filters["all_projects"] = False
```
Comment (shruthilayaj) [inline src/sentry/api/serializers/rest_framework/dashboard.py:335]: nit: just for consistency, should we avoid modifying instance.filter directly here and do
`filters["all_projects"] = False` and let filters be set on instances in line 347?

File: src/sentry/api/serializers/rest_framework/dashboard.py [hunk 2]
```diff
@@ -324,10 +325,16 @@ def update_dashboard_filters(self, instance, validated_data):
...
+                instance.projects.set(validated_data["projects"])
```
Comment (shruthilayaj) [inline src/sentry/api/serializers/rest_framework/dashboard.py:336]: Do we need to clear instance.projects if all projects is selected?

**Review by shruthilayaj — APPROVED** [review]

File: src/sentry/api/serializers/rest_framework/dashboard.py (hunk 1 above)
Comment (narsaynorath) [reply src/sentry/api/serializers/rest_framework/dashboard.py]: Thanks for calling that out. It was why I moved `filters` up but I must've gone and tried something else at the time. 👍

File: src/sentry/api/serializers/rest_framework/dashboard.py (hunk 2 above)
Comment (narsaynorath) [reply src/sentry/api/serializers/rest_framework/dashboard.py:337]: Yes! Good catch. I've added an `instance.projects.clear()` when setting `all_projects` to `True`

**Pull request merged**